src/Bicep.Core/Semantics/Namespaces/ThirdPartyNamespaceType.cs

```diff
@@ -17,14 +18,32 @@ public static class ThirdPartyNamespaceType
 
         public static NamespaceType Create(string name, string aliasName, IResourceTypeProvider resourceTypeProvider)
         {
-            return new NamespaceType(
-                aliasName,
-                new NamespaceSettings(
+            var namespaceSettings = new NamespaceSettings(
                     IsSingleton: true,
                     BicepProviderName: name,
                     ConfigurationType: null,
                     ArmTemplateProviderName: name,
-                    ArmTemplateProviderVersion: resourceTypeProvider.Version),
+                    ArmTemplateProviderVersion: resourceTypeProvider.Version);
+
+            if (resourceTypeProvider is ThirdPartyResourceTypeProvider thirdPartyProvider)
+            {
+                var namespaceConfig = thirdPartyProvider.GetNamespaceConfiguration();
+
+                if (namespaceConfig is not null)
+                {
+                    namespaceSettings = new NamespaceSettings(
+                        IsSingleton: namespaceConfig.IsSingleton,
```

Comment:
Should both BicepProviderName and ArmTemplateProviderName be set to namespaceConfig.Name?

Reply:
I think so for now. In the long run, we may need to rethink this, but I think out of scope for now.